Source/Bogus/Extensions/ExtensionsForItaly/CodiceFiscaleBuilder.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Bogus.Extensions
+{
+    /// <summary>
+    /// This class implements the principal routines of the Italian fiscal code,
+    /// used to unambiguously identify individuals residing in Italy.
+    /// <see cref="https://en.wikipedia.org/wiki/Italian_fiscal_code_card"/>
+    /// </summary>
+    internal static class CodiceFiscaleBuilder
+    {
+        /// <summary>
+        ///   Generates an Italian Fiscal Code
+        /// </summary>
+        /// <param name="lastName">Lastname of the holder</param>
+        /// <param name="firstName">Firstname of the holder</param>
+        /// <param name="birthday">Birthday of the holder</param>
+        /// <param name="male">Indicates whether the holder is male</param>
+        /// <param name="validChecksum">
+        ///   Indicates whether the generated Fiscal Code has a valid checksum or not
+        /// </param>
+        /// <returns>The generated Fiscal Code</returns>
+        public static string Build(
+            string lastName,
+            string firstName,
+            DateTime birthday,
+            bool male,
+            bool validChecksum = true)
+        {
+            var faker = new Faker();
```

Comment:
Great work @supix, you've done an amazing job here. Your tests look great. My only concern is this line above. `var faker = new Faker();`, do we need this here? Seems like the variable is unused here. I think it looks safe to delete this line.

We need to be careful newing up `new Faker()` inside **Extension Methods** because using `new Faker()` in **Extension Methods** can lead the possibility of breaking Bogus' deterministic behavior for a given seed. If you need randomness in the future, it is best to draw randomness from the `Random` property of a `DataSet` or `Person` instead of `new Faker()`.

Reply:
Ops... fixed! :-)